x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,531 @@
+  projectName: string,
+  runnerId: string,
+  apiKey: string,
+  ftrConfig: SecuritySolutionDescribeBlockFtrConfig,
+  onEarlyExit: (msg: string) => void
+): Promise<Environment> {
+  log.info(`${runnerId}: Creating environment ${projectName}...`);
+  const environment = {} as Environment;
+  const body = {
+    name: projectName,
+    region_id: DEFAULT_REGION,
+  } as CreateEnvironmentRequestBody;
+
+  const productTypes: ProductType[] = [];
+  ftrConfig?.productTypes?.forEach((t) => {
+    productTypes.push(t as ProductType);
+  });
+  if (productTypes.length > 0) body.product_types = productTypes;
+
+  await axios
+    .post(`${baseUrl}/api/v1/serverless/projects/security`, body, {
+      headers: {
+        Authorization: `ApiKey ${apiKey}`,
+        'Content-Type': 'application/json',
+      },
+    })
+    .then((response) => {
+      environment.name = response.data.name;
+      environment.id = response.data.id;
+      environment.region = response.data.region_id;
+      environment.es_url = `${response.data.endpoints.elasticsearch}:443`;
+      environment.kb_url = `${response.data.endpoints.kibana}:443`;
+      environment.product = response.data.type;
+    })
+    .catch((error) => {
+      onEarlyExit(`${error.code}:${error.data}`);
+    });
+  return environment;
+}
+
+// Method to invoke the delete environment API for serverless.
+async function deleteEnvironment(
+  baseUrl: string,
+  projectId: string,
+  projectName: string,
+  runnerId: string,
+  apiKey: string,
+  onEarlyExit: (msg: string) => void
+) {
+  await axios
+    .delete(`${baseUrl}/api/v1/serverless/projects/security/${projectId}`, {
+      headers: {
+        Authorization: `ApiKey ${apiKey}`,
+      },
+    })
+    .then((response) => {
+      log.info(`${runnerId} : Environment ${projectName} was successfully deleted...`);
+    })
+    .catch((error) => {
+      onEarlyExit(`${error.code}:${error.data}`);
+    });
+}
+
+// Method to reset the credentials for the created environment.
+async function resetCredentials(
+  baseUrl: string,
+  environmentId: string,
+  runnerId: string,
+  apiKey: string
+): Promise<Credentials> {
+  log.info(`${runnerId} : Reseting credentials`);
+  const credentials = {} as Credentials;
+
+  await poll(async () => {
+    return axios
+      .post(
+        `${baseUrl}/api/v1/serverless/projects/security/${environmentId}/_reset-credentials`,
+        {},
+        {
+          headers: {
+            Authorization: `ApiKey ${apiKey}`,
+          },
+        }
+      )
+      .then((response) => {
+        credentials.password = response.data.password;
+        credentials.username = response.data.username;
+      })
+      .catch((error) => {
+        throw new Error(`${error.code}:${error.data}`);
+      });
+  });
+  return credentials;
+}
+
+// Wait until elasticsearch status goes green
+async function waitForEsStatusGreen(esUrl: string, auth: string, runnerId: string) {
+  await poll(async () => {
+    await axios
```

Comment:
The same comments as for `createEnvironment` are applicable here.

Reply:
Fixed with the commit https://github.com/elastic/kibana/pull/169311/commits/597d8c197258b5b365016979f23fdbb568abea4e

Reply:
Polling is removed. TRy catch introduced.